src/test/regress/sql/upgrade_basic_after.sql

```diff
@@ -8,8 +8,18 @@ SELECT nextval('pg_dist_placement_placementid_seq') = MAX(placementid)+1 FROM pg
 SELECT nextval('pg_dist_groupid_seq') = MAX(groupid)+1 FROM pg_dist_node;
 SELECT nextval('pg_dist_node_nodeid_seq') = MAX(nodeid)+1 FROM pg_dist_node;
 SELECT nextval('pg_dist_colocationid_seq') = MAX(colocationid)+1 FROM pg_dist_colocation;
-SELECT nextval('pg_dist_operationid_seq') = MAX(operation_id)+1 FROM pg_dist_cleanup;
-SELECT nextval('pg_dist_cleanup_recordid_seq') = MAX(record_id)+1 FROM pg_dist_cleanup;
+SELECT
+    CASE WHEN MAX(operation_id) IS NULL
+    THEN true
+    ELSE nextval('pg_dist_operationid_seq') = MAX(operation_id)+1
+    END AS check_operationid
+    FROM pg_dist_cleanup;
+SELECT
+    CASE WHEN MAX(record_id) IS NULL
+    THEN true
+    ELSE nextval('pg_dist_cleanup_recordid_seq') = MAX(record_id)+1
+    END AS check_recordid
+    FROM pg_dist_cleanup;
```

Comment:
this change seems unrelated on first sight, why do we need to rewrite these tests now?

Instead, I see some references to pg_dist_cleanup in the original PR, are we uncovering/hiding an upgrade bug here? Shouldn't the sequences retain the correct value by accompanying logic in `citus_(prepare|finish)_pg_upgrade`?

Reply:
this test file is called in two schedules: citus upgrade and the pg upgrade one.
We create cleanup record with the test in this pr in citus upgrade schedule, not in pg upgrade schedule.
In pg upgrade schedule, because no records are created, the checks for these sequences return null, but in the citus upgrade schedule, they return true. That's why I changed these sequence checks so that they are expected to be either null, or max+1.

Reply:
> In pg upgrade schedule, because no records are created, the checks for these sequences return null

On deeper inspection I do see that, even though the tests look very much like the tests above, the output used to be `NULL` for these two lines, compared to the expected true value.

This is probably ok, but the deviation from the tests above warrants a comment describing the situation where the `MAX(operation_id) IS NULL`. Given the precedent above where sequences are generally tested the simple way, like before this PR -albeit that test was wrong for these sequences and should not have been added given the wrong output.

However, I don't think it is easy or possible, but having `pg_dist_cleanup` populated during postgres upgrades would be even better. Although I don't think that would be a valid scenario, but that would be great to get affirmed. (do understand that I don't really understand the initial change you were making - maybe it's better to involve the people that reviewed https://github.com/citusdata/citus/pull/6494)